### PR TITLE
Fix double tap zoom some images

### PR DESCRIPTION
**Pull request by vladrevers**
https://github.com/FossifyOrg/Gallery/issues/40

### CODE REVIEW

File: library/src/main/kotlin/com/davemorrissey/labs/subscaleview/SubsamplingScaleImageView.kt [hunk 1]
```diff
@@ -628,14 +628,17 @@ open class SubsamplingScaleImageView @JvmOverloads constructor(context: Context,
             val targetScale = if (zoomIn) doubleTapZoomScale else getFullScale()
             AnimationBuilder(sCenter!!, targetScale).start()
         } else {
-            val targetScale = if (zoomIn && scale != 1f) doubleTapZoomScale else getFullScale()
+            var targetScale = if (zoomIn && scale != 1f) doubleTapZoomScale else getFullScale()
             if (scale != 1f) {
                 if (zoomIn) {
                     AnimationBuilder(sCenter!!, targetScale).start()
                 } else {
-                    AnimationBuilder(sCenter!!, 1f).start()
```
Comment (vladrevers) [inline library/src/main/kotlin/com/davemorrissey/labs/subscaleview/SubsamplingScaleImageView.kt:636]: Necessary for normal un-zoom to original view in small images.

File: library/src/main/kotlin/com/davemorrissey/labs/subscaleview/SubsamplingScaleImageView.kt [hunk 2]
```diff
@@ -628,14 +628,17 @@ open class SubsamplingScaleImageView @JvmOverloads constructor(context: Context,
...
+                    AnimationBuilder(sCenter!!, Math.max(1f, targetScale)).start()
```
Comment (vladrevers) [inline library/src/main/kotlin/com/davemorrissey/labs/subscaleview/SubsamplingScaleImageView.kt:636]: Necessary for normal un-zoom to original view in small images. For large images targetScale values are less than 1.

**Comment by naveensingh** [comment]
Looks good to me, thanks!

**Comment by naveensingh** [comment]
Actually, it still glitches for smaller images with the same aspect ratio as the device. I guess it'll be fixed later.

**Pull request merged**